src/main/java/io/jenkins/plugins/junit/checks/JUnitChecksPublisher.java

```diff
@@ -0,0 +1,143 @@
+package io.jenkins.plugins.junit.checks;
+
+import edu.hm.hafner.util.VisibleForTesting;
+import hudson.model.TaskListener;
+import hudson.tasks.junit.CaseResult;
+import hudson.tasks.junit.TestResultAction;
+import hudson.tasks.junit.TestResultSummary;
+import io.jenkins.plugins.checks.api.ChecksConclusion;
+import io.jenkins.plugins.checks.api.ChecksDetails;
+import io.jenkins.plugins.checks.api.ChecksOutput;
+import io.jenkins.plugins.checks.api.ChecksPublisher;
+import io.jenkins.plugins.checks.api.ChecksPublisherFactory;
+import io.jenkins.plugins.checks.api.ChecksStatus;
+import io.jenkins.plugins.util.JenkinsFacade;
+import java.util.List;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+@Restricted(NoExternalUse.class)
+public class JUnitChecksPublisher {
+    public static final String SEPARATOR = ", ";
+    private final TestResultAction action;
+    private final JenkinsFacade jenkinsFacade;
+    private final TestResultSummary summary;
+
+    public JUnitChecksPublisher(final TestResultAction action, final TestResultSummary summary) {
+        this(action, summary, new JenkinsFacade());
+    }
+
+    @VisibleForTesting
+    JUnitChecksPublisher(final TestResultAction action, final TestResultSummary summary, final JenkinsFacade jenkinsFacade) {
+        this.jenkinsFacade = jenkinsFacade;
+        this.action = action;
+        this.summary = summary;
+    }
+
+    public void publishChecks(TaskListener listener) {
+        ChecksPublisher publisher = ChecksPublisherFactory.fromRun(action.run, listener);
+        publisher.publish(extractChecksDetails());
+    }
+
+    @VisibleForTesting
+    ChecksDetails extractChecksDetails() {
+        String text = extractChecksText();
+        System.out.println(text);
+        ChecksOutput output = new ChecksOutput.ChecksOutputBuilder()
+                .withTitle(extractChecksTitle())
+                .withSummary("<sub>Send us [feedback](https://github.com/jenkinsci/junit-plugin/issues)")
```

Comment:
I was wondering about maybe having a Feedback <- survey, and a bug report link, thoughts?

Reply:
Whatever works best for you. Just GitHub issues might be enough for developers, but a survey is better for common users